xwiki-platform-core/xwiki-platform-csrf/src/main/java/org/xwiki/csrf/internal/DefaultCSRFToken.java

```diff
@@ -191,11 +195,15 @@ public String getResubmissionURL()
     {
         String query = "resubmit=" + urlEncode(getRequestURI());
 
-        // back URL is the URL of the document that was about to be modified, so in most
+        // backURL is the URL of the document that was about to be modified, so in most
         // cases we can redirect back to the correct document (if the user clicks "no")
-        String backUrl = getDocumentURL(this.docBridge.getCurrentDocumentReference(), null);
+        DocumentReference currentDocRef = this.docBridge.getCurrentDocumentReference();
+        // If the current document is null (on the login page for example), let's redirect to the default page
```

Comment:
Wouldn't be cleaner to get the default page reference by accessing the current wiki descriptor? Something like: 
```
@Inject
private WikiDescriptorManager wikiDescriptorManager;

....

this.wikiDescriptorManager.getCurrentWikiDescriptor().getMainPageReference();
```

Reply:
I'll try that.

Reply:
Yes, being redirected to the current wiki home page will probably be less of a WTF than ending up on a completely different wiki :)